app/services/file_mover.rb

```diff
@@ -0,0 +1,45 @@
+# Copyright 2011-2024, The Trustees of Indiana University and Northwestern
+#   University.  Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+# 
+# You may obtain a copy of the License at
+# 
+# http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software distributed
+#   under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+#   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+#   specific language governing permissions and limitations under the License.
+# ---  END LICENSE_HEADER BLOCK  ---
+
+class FileMover
+  class << self
```

Comment:
I like the refactoring to create this new `FileMover` service.  What do you think about making a convenience method to simplify the calls to it?
```suggestion
  class << self
    def move(source, dest, method: nil)
      new(source, dest, method).move
    end
  end
  
  def initialize(source, dest, method: nil)
    @source = source
    @dest = dest
    @method = method
  end
  
  def move
    send(method, @source, @dest)
  end
  
  def method
    @method || "#{@source.uri.scheme}_to_#{@dest.uri.scheme}".to_sym
  end
  
  private
```

Reply:
I like that idea. Should clean some things up nicely.